test-d/set-mutable.ts

```diff
@@ -0,0 +1,18 @@
+import {expectType, expectError} from 'tsd';
+import {SetMutable} from '..';
+
+// Update one required and one optional to required.
+declare const variation1: SetMutable<{readonly a: number; b: string; readonly c: boolean}, 'b' | 'c'>;
+expectType<{readonly a: number; b: string; c: boolean}>(variation1);
+
+// Update two optional to required.
+declare const variation2: SetMutable<{readonly a: number; readonly b: string; readonly c: boolean}, 'a' | 'b'>;
+expectType<{a: number; b: string; readonly c: boolean}>(variation2);
+
+// Three required remain required.
+declare const variation3: SetMutable<{a: number; b: string; c: boolean}, 'a' | 'b' | 'c'>;
+expectType<{a: number; b: string; c: boolean}>(variation3);
+
+// Fail if type changes even if optional is right.
+declare const variation4: SetMutable<{readonly a: number; b: string; readonly c: boolean}, 'b' | 'c'>;
+expectError<{readonly a: boolean; b: string; c: boolean}>(variation4);
```

Comment:
I think you have a copy and paste mistake with the comments here. They are identical to the ones in https://github.com/sindresorhus/type-fest/blob/master/test-d/set-required.ts and doesn't make much sense in this context, making me somewhat confused at first 😉

Reply:
Oh yeah, my bad! Nice catch 🥇